src/components/pages/roster-page/roster-page.tsx

```diff
@@ -142,16 +183,134 @@ export const RosterPage = () => {
+    setSelectedRosterEntry(rosterEntry);
+    setEditRosterEntryDialogProps({
+      open: true,
+      orgId,
+      rosterColumnInfos,
+      rosterEntry,
+      onClose: async () => {
+        setEditRosterEntryDialogProps({ open: false });
+        setSelectedRosterEntry(undefined);
+        await initializeTable();
+      },
+      onError: (message: string) => {
+        setAlertDialogProps({
+          open: true,
+          title: 'Edit Roster Entry',
+          message: `Unable to edit roster entry: ${message}`,
+          onClose: () => { setAlertDialogProps({ open: false }); },
+        });
+      },
+    });
   };
 
-  useEffect(() => { initializeTable().then(); }, []);
+  const addButtonClicked = async () => {
+    setEditRosterEntryDialogProps({
+      open: true,
+      orgId,
+      rosterColumnInfos,
+      onClose: async () => {
+        setEditRosterEntryDialogProps({ open: false });
+        setSelectedRosterEntry(undefined);
+        await initializeTable();
+      },
+      onError: (message: string) => {
+        setAlertDialogProps({
+          open: true,
+          title: 'Add Roster Entry',
+          message: `Unable to add roster entry: ${message}`,
+          onClose: () => { setAlertDialogProps({ open: false }); },
+        });
+      },
+    });
+  };
+
+  function deleteButtonClicked(rosterEntry: ApiRosterEntry) {
+    setSelectedRosterEntry(rosterEntry);
+    setDeleteRosterEntryDialogOpen(true);
+  }
+
+  const deleteRosterEntry = async () => {
+    try {
+      setDeleteRosterEntryLoading(true);
+      await axios.delete(`api/roster/${orgId}/${selectedRosterEntry!.edipi}`);
+    } catch (error) {
+      let message = 'Internal Server Error';
+      if (error.response?.data?.errors && error.response.data.errors.length > 0) {
+        message = error.response.data.errors[0].message;
+      }
+      setAlertDialogProps({
+        open: true,
+        title: 'Delete Roster Entry',
+        message: `Unable to delete roster entry: ${message}`,
+        onClose: () => { setAlertDialogProps({ open: false }); },
+      });
+    } finally {
+      setDeleteRosterEntryLoading(false);
+      setDeleteRosterEntryDialogOpen(false);
+      setSelectedRosterEntry(undefined);
+      await initializeTable();
+    }
+  };
+
+  const cancelDeleteRosterEntryDialog = () => {
+    setDeleteRosterEntryDialogOpen(false);
+    setSelectedRosterEntry(undefined);
+  };
+
+  useEffect(() => { initializeTable().then(); }, [initializeTable]);
+
+  const buildColumnHeaders = () => {
+    const columns = rosterColumnInfos?.slice(0, maxNumColumnsToShow);
+    return columns?.map(columnInfo => (
+      <TableCell key={columnInfo.name}>{columnInfo.displayName}</TableCell>
+    ));
+  };
+
+  const buildTableRows = () => {
+    const columns = rosterColumnInfos?.slice(0, maxNumColumnsToShow);
+
+    return rows.map(row => (
+
+      // TODO - what if we don't have edipi? What should we make this key?
+      <TableRow key={row.edipi}>
```

Comment:
Can we always assume that we have an edipi parameter? If we don't, I'm not too sure what to put for the key here.